include/actor_types.h

```diff
@@ -2,8 +2,17 @@
 #include <macros.h>
 #include <common_structs.h>
 
-// Made this because there will likely be a lot of structs in this file that will be compiled into one eventually.
-// This keeps the other files cleaner.
+/*
+The actor list (gActorList) should be understood to be populated by generic struct Actor elements.
+However, for human readability, many functions interacting with actor list elements expect one of the many
+specialized types found in this file.
+
+Note that specialized types must be the same size as a plain Actor. Don't be mislead into thinking that
+because its a separate type that it can modified separately from plain Actor. If you modify/add an actor type
+and its size is different from plain Actor's, you WILL run into buggy (potentially crash inducing) behaviour.
+This includes having the member at byte 0x30 be an UnkActorInner struct, the member at 0x00 be the `type`
+of the actor, and the member at 0x02 be the `flags`. Other members are more flexible
+*/
```

Comment:
I really like the comments here. Couple wording ideas:

Actors are populated in gActorList using generic Actor structs. (not quite sure how 'elements' works its way in this version)
Some actors require a specialized struct.

`This includes having the member at`... Maybe something like:
The struct members are customization so long the below spec is followed:

Required members:
    0x00 -> type   
    0x02 -> flags    
    0x30 -> UnkActorInner